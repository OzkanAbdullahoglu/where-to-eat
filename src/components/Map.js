import React, { Component } from 'react';
import { connect } from 'react-redux';
import { compose } from 'recompose';

import SearchMap from './SearchMap';
import GoogleMapComponent from './GoogleMapComponent';
import LeftPanel from './LeftPanel';
import { getFetchedData } from '../reducers';

class Map extends Component {
  state = {
    // default center as a start
    center: { lat: 41.0082376, lng: 28.9783589 },
    // default mapping zoom
    zoom: 12,
  };

  /**
   *  @description updating center of the map by manipulating the state of center
   *  @param {number} newlat- we  are using newLat and newLng to update lat and lng states
   *  @param {number} newlng
   *  updating zoom of map by manipulating the state of zoom
   *  to reset all venue IDs for a fresh start?????????*
   *  calling getParameters function to fetch new locations????????
   */
  markerLocationUpdate = (newLat, newLng) => {
    this.setState({ center: { lat: newLat, lng: newLng } });
    this.setState({ zoom: 15 });
  };

  zoomToUserLocation = () => {
    if (navigator.geolocation) {
      this.geoSuccess = (position) => {
        const newLat = position.coords.latitude;
        const newLng = position.coords.longitude;
        this.setState({
          lat: newLat,
          lng: newLng,
        });
        this.markerLocationUpdate(newLat, newLng);
      };
      this.geoError = (error) => {
        switch (error.code) {
          case 0:
            alert(
              'Unknown error occured while attempting to identify user location'
            );
            break;
          case 1:
            alert('Permission denied!');
            break;
          case 2:
            alert('Position is unavailable');
            break;
          case 3:
            alert('Operation time out');
            break;
          default:
            console.log(error.code);
            break;
        }
      };
      navigator.geolocation.getCurrentPosition(this.geoSuccess, this.geoError);
    } else {
      console.log('Geolocation is not supported for this Browser/OS.');
    }
  };
  render() {
    const { lat, lng } = this.state.center;
    return (
      <div className="main-container" role="main">
        <div
          id="map-container"
          role="application"
          aria-label="Google Map Container"
        >
          <LeftPanel
            lat={lat}
            lng={lng}
            userLocationUpdate={this.zoomToUserLocation}
          />
          <GoogleMapComponent
            center={this.state.center}
            zoom={this.state.zoom}
          />
          <SearchMap markerLocationTrigger={this.markerLocationUpdate} />
        </div>
      </div>
    );
  }
}

const mapStateToProps = (store) => ({
  isFetchedData: getFetchedData(store),
});

const withRedux = connect(mapStateToProps);

export default compose(withRedux)(Map);
