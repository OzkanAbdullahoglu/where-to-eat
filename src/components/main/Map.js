/* eslint-disable linebreak-style */
import React, { Component } from 'react';
import whyDidYouRender from '@welldone-software/why-did-you-render/dist/no-classes-transpile/umd/whyDidYouRender';
import { connect } from 'react-redux';
import { compose } from 'recompose';
import PropTypes from 'prop-types';

import SearchMap from './SearchMap';
import GoogleMapComponent from './GoogleMapComponent';
import LeftPanel from './LeftPanel';
import { getFetchedData, getSearchBoxStatus, mapActions } from '../../reducers';

whyDidYouRender(React, {
  onlyLogs: true,
  titleColor: 'green',
  diffNameColor: 'aqua',
});

class Map extends Component {
/* static whyDidYouRender = true;*/
  state = {
    // default center as a start incase we can't get user location at the begining
    center: { lat: 37.09024, lng: -95.712891 },
    // default mapping zoom
    zoom: 12,
  };

  componentDidMount() {
    this.zoomToUserLocation();
    /* this.props.setDefault();*/
  }
  shouldComponentUpdate(nextProps, nextState) {
    const { center, zoom } = this.state;
    return (
      center !== nextState.center ||
        zoom !== nextState.zoom ||
        this.props.isFetchedData !== nextProps.isFetchedData
    );
  }

  /**
   *  @description updating center of the map by manipulating the state of center
   *  @param {number} newlat- we  are using newLat and newLng to update lat and lng states
   *  @param {number} newlng
   *  updating zoom of map by manipulating the state of zoom
   */
  markerLocationUpdate = (newLat, newLng) => {
    this.setState({ zoom: 15 });
    this.setState({ center: { lat: newLat, lng: newLng } });
  };

  zoomToUserLocation = () => {
    if (navigator.geolocation) {
      this.geoSuccess = (position) => {
        const newLat = position.coords.latitude;
        const newLng = position.coords.longitude;
        /* this.setState({
          lat: newLat,
          lng: newLng,
        });*/

        this.markerLocationUpdate(newLat, newLng);
      };
      this.geoError = (error) => {
        switch (error.code) {
          case 0:
            alert(
              'Unknown error occured while attempting to identify user location'
            );
            break;
          case 1:
            alert('Permission denied!');
            break;
          case 2:
            alert('Position is unavailable');
            break;
          case 3:
            alert('Operation time out');
            break;
          default:
            console.log(error.code);
            break;
        }
      };
      navigator.geolocation.getCurrentPosition(this.geoSuccess, this.geoError);
    } else {
      console.log('Geolocation is not supported for this Browser/OS.');
    }
  };

  render() {
    const { lat, lng } = this.state.center;
    return (
      <div className="main-container" role="main">
        <div
          id="map-container"
          role="application"
          aria-label="Google Map Container"
        >
          <LeftPanel
            lat={lat}
            lng={lng}
            userLocationUpdate={this.zoomToUserLocation}
          />
          <GoogleMapComponent
            center={this.state.center}
            zoom={this.state.zoom}
          />
          <SearchMap
            markerLocationTrigger={this.markerLocationUpdate}
          />
        </div>
      </div>
    );
  }
}

Map.propTypes = {
  isFetchedData: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
  ]),
};

const mapStateToProps = (store) => ({
  isFetchedData: getFetchedData(store),
  isSearchActivated: getSearchBoxStatus(store),
});

const withRedux = connect(mapStateToProps, { ...mapActions });

export default compose(withRedux)(Map);
