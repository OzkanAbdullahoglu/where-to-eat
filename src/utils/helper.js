
export const openHoursManage = (obj) => {
  let willOpen = '';
  let willClosed = '';
  let openUntil = '';
  const date = new Date();
  const today = date.getDay();
  const time = parseInt(
    `${JSON.stringify(date.getHours())}${JSON.stringify(date.getMinutes())}`, 10
  );

  if (obj === undefined || obj.open[today] === undefined) {
    return '';
  }
  const closeTimeInt = parseInt(obj.open[today].end, 10);
  const openTime = `${obj.open[today].start.slice(0, 2)}:${obj.open[today].start.slice(2)}`;
  const closeTime = `${obj.open[today].end.slice(0, 2)}:${obj.open[
    today
  ].end.slice(2)}`;
  if (closeTimeInt - time < 100) {
    willClosed = `Close soon ${closeTime}`;
  } else if (time > closeTimeInt || time < closeTimeInt) {
    willOpen = `Opens at ${openTime}`;
  } else {
    openUntil = `Open until ${closeTime}`;
  }
  return (
    willClosed || willOpen || openUntil
  );
};

export const getCategoriesTitle = (arr) => {
  const titleArr = arr.map((obj) => obj.title);
  const concatenateTitles = titleArr.reduce(
    (accTitles, singleTitle) => `${accTitles}, ${singleTitle}`
  );
  return concatenateTitles;
};
