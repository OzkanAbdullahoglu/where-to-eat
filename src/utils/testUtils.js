/* eslint-disable linebreak-style */
import { createStore, applyMiddleware } from 'redux';
import { middlewares } from '../store';
import rootReducer from '../reducers';


export const storeFactory = (initalState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
  return createStoreWithMiddleware(rootReducer, initalState);
}
;
