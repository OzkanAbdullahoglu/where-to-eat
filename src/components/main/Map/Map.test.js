/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';

import { storeFactory } from '../../../utils/testUtils';
import Map from './Map';

describe('Map', () => {
  const store = storeFactory();
  const map = shallow(<Map store={store} />).dive().dive();
  it('renders map component correctly', () => {
    expect(map).toMatchSnapshot();
  });
  it('contains a Left Panel Component', () => {
    expect(map.find('Connect(LeftPanel)').exists()).toBe(true);
  });
  it('contains a Google Map Component', () => {
    expect(map.find('Connect(withProps(lifecycle(withScriptjs(withGoogleMap(Component)))))').exists()).toBe(true);
  });
  it('contains a Search Box Component', () => {
    expect(map.find('Connect(SearchMap)').exists()).toBe(true);
  });
  it('has an initial center with a lat 37.09024 and a lng -95.712891, and zoom 12', () => {
    const lat = 37.09024;
    const lng = -95.712891;
    expect(map.state()).toEqual({ center: { lat, lng }, zoom: 12 });
  });
  describe('updating coordinates and zoom', () => {
    const newLat = 25.25;
    const newLng = 50.50;
    beforeEach(() => {
      map.instance().markerLocationUpdate(newLat, newLng);
    });
    it('updates coordinates and zoom', () => {
      expect(map.state()).toEqual({ center: { lat: newLat, lng: newLng }, zoom: 15 });
    });
  });
});

