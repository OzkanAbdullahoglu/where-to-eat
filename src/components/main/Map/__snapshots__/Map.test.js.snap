// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map renders map component correctly 1`] = `
<div
  className="main-container"
  role="main"
>
  <div
    aria-label="Google Map Container"
    id="map-container"
    role="application"
  >
    <Connect(BareLeftPanel)
      lat={37.09024}
      lng={-95.712891}
      userLocationUpdate={[Function]}
    />
    <Connect(withProps(lifecycle(withScriptjs(withGoogleMap(Component)))))
      center={
        Object {
          "lat": 37.09024,
          "lng": -95.712891,
        }
      }
      zoom={12}
    />
    <Connect(SearchMap)
      markerLocationTrigger={[Function]}
    />
  </div>
</div>
`;
